src/tools/bzlmod/utils.bzl

```diff
@@ -109,6 +141,83 @@ def parse_http_artifacts(ctx, lockfile_path, required_repos):
 
     return http_artifacts
 
+BCR_URL_SCHEME = "https://bcr.bazel.build/modules/{name}/{version}/{file}"
+
+def parse_registry_files(ctx, lockfile_path, module_files):
+    """Parses the registry files referenced by the given lockfile and returns them in http_file form.
+
+    Args:
+        ctx: the repository / module extension ctx object.
+        lockfile_path: The path of the lockfile to extract the registry files from.
+        module_files: The paths of non-registry module files to use during fake module resolution.
+
+    Returns:
+        A list of http artifacts in the form of
+        [{"sha256": <sha256 value>, "url": <url>}, ...]
+    """
+    lockfile = json.decode(ctx.read(lockfile_path))
+    registry_file_hashes = lockfile.get("registryFileHashes", {})
+    if registry_file_hashes:
+        return [
+            {"sha256": sha256, "url": url}
+            for url, sha256 in registry_file_hashes.items()
+        ]
+
+    # TODO: Remove this branch after Bazel is built with 7.2.0.
```

Comment:
nit - s/this branch/the following code

also maybe mention that we can remove the Buildozer dep too